docs/docs/dev_guide/profiles/cprofile.md

```diff
@@ -0,0 +1,95 @@
+          ps = pstats.Stats(pr, stream=s).sort_stats(SortKey.TIME)
+          ps.print_stats(10)
+          return result
+```
+
+```python
+""" This is with using the `cProfile` profiler without the ContextManager Object """
+import cProfile
+import pstats
+from pstats import SortKey
+
+pr = cProfile.Profile()
+pr.enable()
+with plugin_result(plugin_name, write_design_doc_to_directory) as result:
+     s = io.StringIO() // this is used to capture any IO streams
+     sortby = SortKey.CUMULATIVE
+     ps = pstats.Stats(pr, stream=s).sort_stats(sortby)
+     ps.print_stats()
+     return result
+```
+
+So lets break this down a bit:
+When you don't sort your results you can get a lot of noise from some functions and methods that are being called and can really
+disorient and add to the confusion of what is going on with the plugin. The above code was run for the `gen_design_doc_impl.py` file
+and within the results as seen with `ps.print_stats(10)` we are filtering the top ten from the line before its sort.
+
+Through the sort function, and filtering the sort you can see the heavy offenders of the plugin that is being profiled.
+
+The output from this run will have something similar to the below:
+
+```bash
+aac gen-design-doc ./model/alarm_clock/alarm_clock.yaml ../../../temp/
+         4040225 function calls (3966008 primitive calls) in 1.194 seconds
+
+   Ordered by: internal time
+   List reduced from 1186 to 10 due to restriction <10>
+
+   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
+    19692    0.085    0.000    0.085    0.000 /Users/charles.blackard/venvs/AaC/python/src/aac/lang/language_context.py:361(<listcomp>)
+   107541    0.080    0.000    0.083    0.000 /Users/charles.blackard/venvs/AaC/python/venv/lib/python3.9/site-packages/yaml/reader.py:99(forward)
+     8986    0.062    0.000    0.062    0.000 {method 'flush' of '_io.TextIOWrapper' objects}
+     8489    0.054    0.000    0.219    0.000 /Users/charles.blackard/venvs/AaC/python/venv/lib/python3.9/site-packages/yaml/scanner.py:1270(scan_plain)
+   452287    0.047    0.000    0.047    0.000 /Users/charles.blackard/venvs/AaC/python/venv/lib/python3.9/site-packages/yaml/reader.py:87(peek)
+       53    0.038    0.001    0.038    0.001 <frozen importlib._bootstrap_external>:1053(open_resource)
+   114364    0.030    0.000    0.054    0.000 /Users/charles.blackard/venvs/AaC/python/venv/lib/python3.9/site-packages/yaml/scanner.py:145(need_more_tokens)
+    13605    0.028    0.000    0.028    0.000 /Users/charles.blackard/venvs/AaC/python/src/aac/lang/language_context.py:305(<listcomp>)
+    22005    0.027    0.000    0.074    0.000 /Users/charles.blackard/venvs/AaC/python/venv/lib/python3.9/site-packages/yaml/scanner.py:1311(scan_plain_spaces)
+     8986    0.026    0.000    0.054    0.000 /opt/homebrew/Cellar/python@3.9/3.9.14/Frameworks/Python.framework/Versions/3.9/lib/python3.9/logging/__init__.py:282(__init__)
```

Comment:
Anonymized the data a tad. Not a big deal, I'm just against user data in public VCS. Not like we're using our real names and emails for these accounts and commits. 😉  
```suggestion
aac gen-design-doc ./model/alarm_clock/alarm_clock.yaml ../../../temp/
         4040225 function calls (3966008 primitive calls) in 1.194 seconds

   Ordered by: internal time
   List reduced from 1186 to 10 due to restriction <10>

   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
    19692    0.085    0.000    0.085    0.000 /Users/username/venvs/AaC/python/src/aac/lang/language_context.py:361(<listcomp>)
   107541    0.080    0.000    0.083    0.000 /Users/username/venvs/AaC/python/venv/lib/python3.9/site-packages/yaml/reader.py:99(forward)
     8986    0.062    0.000    0.062    0.000 {method 'flush' of '_io.TextIOWrapper' objects}
     8489    0.054    0.000    0.219    0.000 /Users/username/venvs/AaC/python/venv/lib/python3.9/site-packages/yaml/scanner.py:1270(scan_plain)
   452287    0.047    0.000    0.047    0.000 /Users/username/venvs/AaC/python/venv/lib/python3.9/site-packages/yaml/reader.py:87(peek)
       53    0.038    0.001    0.038    0.001 <frozen importlib._bootstrap_external>:1053(open_resource)
   114364    0.030    0.000    0.054    0.000 /Users/username/venvs/AaC/python/venv/lib/python3.9/site-packages/yaml/scanner.py:145(need_more_tokens)
    13605    0.028    0.000    0.028    0.000 /Users/username/venvs/AaC/python/src/aac/lang/language_context.py:305(<listcomp>)
    22005    0.027    0.000    0.074    0.000 /Users/username/venvs/AaC/python/venv/lib/python3.9/site-packages/yaml/scanner.py:1311(scan_plain_spaces)
     8986    0.026    0.000    0.054    0.000 /opt/homebrew/Cellar/python@3.9/3.9.14/Frameworks/Python.framework/Versions/3.9/lib/python3.9/logging/__init__.py:282(__init__)```